### PR TITLE
[MeshMoving] apply rotation and translation as mesh displacement

### PR DESCRIPTION
**Pull request by matekelemen**
**Description**
Add a process that rotates and translates (in that order) nodes and sets the difference to ```MESH_DISPALCEMENT```.

### CODE REVIEW

**Comment by mpentek** [comment]
@KlausBSautter is this something StructuralMechApp could benefit from in case of imposing `DISPLACEMENT`?

**Comment by KlausBSautter** [comment]
> @KlausBSautter is this something StructuralMechApp could benefit from in case of imposing `DISPLACEMENT`?

can you explain more in detail?

**Comment by matekelemen** [comment]
Currently, this process takes the Node's position, transforms it, and sets ```MESH_DISPLACEMENT``` to the difference between the transformed and original positions. I could make this last step adjustable such that, for example, the node's position or ```DISPLACEMENT``` is updated instead of ```MESH_DISPLACEMENT```.

Looking at #8674 however, the question is whether anyone would use such a modified version.

**Comment by mpentek** [comment]
> > @KlausBSautter is this something StructuralMechApp could benefit from in case of imposing `DISPLACEMENT`?
> 
> can you explain more in detail?

Useful for applying some rigid-body-type deformation (either as initial condition or during simulation) in structural mechanics?

File: applications/MeshMovingApplication/custom_processes/impose_mesh_motion_process.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        this->LoadFromEulerAngles(
+            euler_angles,
+            reference_point,
+            translation_vector
+        );
+    } // if rotation_definition == "euler_angles"
+
+    // Rotation definition not implemented
+    else {
+        KRATOS_THROW_ERROR(
```
Comment (philbucher) [inline applications/MeshMovingApplication/custom_processes/impose_mesh_motion_process.cpp]: this macro is deprecated, please use `KRATOS_ERROR` instead

File: applications/MeshMovingApplication/custom_processes/impose_mesh_motion_process.cpp [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    // Parse interval
+    vector_parameter = parameters.GetValue("interval").GetVector();
+    KRATOS_CHECK(vector_parameter.size() == 2);
+    mInterval[0] = vector_parameter[0];
+    mInterval[1] = vector_parameter[1];
+
+    // Parse translation
+    array_1d<double,3> translation_vector;
+    vector_parameter = parameters.GetValue("translation_vector").GetVector();
+    KRATOS_CHECK(vector_parameter.size() == 3);
```
Comment (philbucher) [inline applications/MeshMovingApplication/custom_processes/impose_mesh_motion_process.cpp]: We usually give a meaningful error message, e.g. 
`KRATOS_ERROR << "translation_vector must have a size of 3" << std::endl;`

This is much nicer from a user POV

=> same for other checks

File: applications/MeshMovingApplication/tests/cpp_tests/test_impose_mesh_motion_process.cpp [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 6.00, tolerance);
+
+    node = rModelPart.GetNode(9);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), -2.0, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 6.00, tolerance);
+}
+
+
+KRATOS_TEST_CASE_IN_SUITE(ImposeMeshMotionProcessRotationAxis, MeshMovingApplicationFastSuite)
```
Comment (philbucher) [inline applications/MeshMovingApplication/tests/cpp_tests/test_impose_mesh_motion_process.cpp]: as mentioned in your other PR, I would remove the c++ test if it does the same as the python one
My own rule of thumb is to use C++ for unit tests / things that cannot be tested in python and python for integration tests/larger tests that test entire workflows

My motivation is not to unnecessarily increase compile times

**Review by philbucher — COMMENTED** [review]
ok from my side, only minor comments :+1: 

@mpentek did you / are you going to test this?

**Comment by KlausBSautter** [comment]
Right now, I cannot think of an application in StructMech. But we can discuss this.

**Comment by mpentek** [comment]
> Right now, I cannot think of an application in StructMech. But we can discuss this.

@matekelemen we leave it specific for the MeshMovingApp

**Comment by philbucher** [comment]
for me look s good, @mpentek please approve if its ok for you too

**Review by mpentek — APPROVED** [review]
Thanks

@matekelemen can you please merge and delete?

**Pull request merged**